### ISSUE TITLE
libsumo.simulation.getDistance2D sometimes returns nan on macOS

### ISSUE THREAD

**Issue by behrisch** [eclipse-sumo/sumo#13923]
It is visible in the github actions and affects different python versions see for instance https://github.com/eclipse-sumo/sumo/actions/runs/6457977548 and is non-deterministic. It seems to happen with macOS 12 only.

The call is libsumo.simulation.getDistance2D(pos1geo[0], pos1geo[1],
                                     pos2geo[0], pos2geo[1], isGeo=True))

in the complex/traci/simulation/simulation test

**Comment by RobertHilbrich** [comment]
I am unable to reproduce on macOS 14 Sonoma with Python 3.11. Started complex/traci/simulation/simulation test via texttest 5000x and got the following result: Succeeded: 5000.
 
<img width="645" alt="Bildschirmfoto 2023-10-10 um 17 17 33" src="https://github.com/eclipse-sumo/sumo/assets/1360688/699d8db3-62f8-49c4-a30a-035c2b12d157">

No easy way here to reproduce macOS 12 sigh...

**Comment by RobertHilbrich** [comment]
Of course, we should start the libsumo tests - so I used `texttest -a complex.libsumo` and reran the `complex/traci/simulation/simulation` test 5000 times. However, with the same result.

<img width="760" alt="Bildschirmfoto 2023-10-10 um 20 09 35" src="https://github.com/eclipse-sumo/sumo/assets/1360688/01f02cb6-5b1a-4413-b06c-47c3074ed4d2">
